### DISCUSSION TITLE
how can i use getServerSideProps in _app.js

### DISCUSSION THREAD

**Discussion by komailf67** [vercel/next.js#17442] [Help] [answered]
i want to get data from an Api and send this data as props to all components.
like this codes in index.js

```
const Index = ({ ...props }) => {
  return (
    <React.Fragment>
      <Layout>
        <HomeHeader />
        <MainBody {...props} />
        <Footer />
      </Layout>
    </React.Fragment>
  );
};

export async function getServerSideProps() {
  const mobile = await fetchData("/customers/products/filter/?category=68&limit=10&sort=random&in-stock=true&lang=fa");
  return {
    props: { mobile },
  };
}
```
the mobile is available in MainBody. but i want to access mobile data in all components,

**Comment by jamesmosier** [comment, marked as answer]
Hi @komailf67. There is currently no way to fetch global data inside `_app.js`, aside from using `getInitialProps` which is not recommended [as mentioned in the code comment in the docs](https://nextjs.org/docs/advanced-features/custom-app).

**Reply by komailf67** [reply]
what's your solution for fetching data from server(like getServerSideProps) and  sending it to all components(except redux and localStorage)

**Reply by jamesmosier** [reply]
I personally don't have that need in any of my applications. If I have any global data needs I fetch that client side via the React Context API.

Currently `getInitialProps` in `_app` is the only way to fetch global data for all pages in one place on the server. 

Alternatively [this discussion](https://github.com/vercel/next.js/discussions/17420) mentions another way to fetch the data on each page and then hydrate it in a Context Provider which was an interesting approach.
